### ISSUE TITLE
Feature: Add resetForm function

### ISSUE THREAD

**Issue by niksajanjic** [BenMagyar/form-hooks#9]
Add the ability to reset form back to initial values.

Because API doesn't export setValues I can't manually get them back to inital values.

First idea was to use handleChange for each value, but that function expects event, not string, so I can't just call handleChange('someValue') for each input to reset them back. What I have to do is I have to pass mocked event object with `target.name` to make this work, which I feel is a hack.

Another (hacky) way to reset form would be to pass initial values as props and then reset them back to initial when I need to. Problem is that I have to define them in parent component and create that component just to make this work (which I wouldn't need usually).

I think this should be easy, just call this inside `resetForm`:
```
setErrors({});
setValues(initialValues);
setTouched({});
setIsSubmitting(false);
setSubmitCount(0);
```
Same as you're doing in effect function.

On a side note, I think it would be nice to export all set methods (setValues, setTouched...) so you'll allow customization of values programmatically when needed.

**Comment by BenMagyar** [comment]
Resetting the form is possible through one of the dependency changes. And can be triggered if the `initialValues`, `onSubmit` or `validate` options changes like:

```js
const { errors, touched, values } = useForm(
    { initialValues: {}, ...},
    options => [options.initialValues]
);
```

In hindsight, any value would force a reset there so even if you wanted to have it be reset for some other state change that should be possible through something like `options => [options.initialValues, otherState]`. Is there some additional case where some state change is not what should be causing the form to be reset?

**Comment by niksajanjic** [comment]
Yes, that is one of the ways to reset it, but the problem here is that the only way I can change initialValues is by passing them as props and then changing them from parent component.

Not sure how to change them once I call useForm hook:
```js
const { errors, touched, values } = useForm(
    { initialValues: { firstName: '', lastName: '' }, ...},
    options => [options.initialValues]
);
```
I don't see how to change it after this. Only thing that comes to my mind is to create initialValues as state and then update it later to force a rerender.

One more thing is, I don't really need to change initialValues at all. So, if I have to change them just to reset the form, that feels like a hack. Also, my initialValues are going to be the same at all times, so I have to pass different initialValues (which I don't need to) just to reset form into something which is no more exactly the same (because it can't be the same as it wouldn't trigger reinitialization).

Formik has handleReset just for this use case:
https://jaredpalmer.com/formik/docs/api/formik#handlereset-void

It is usually used in cases when you submit form successfully and then you want to clear it for subsequent submits.

**Comment by BenMagyar** [comment]
That makes sense, it **is** possible that you can do that through another state var, but I agree feels weird in those cases. I'll expose a `resetForm` tonight :+1:

**Comment by niksajanjic** [comment]
Thanks, but don't feel rushed. I just noticed it would be nice to have this function exposed. For now the hack with handleChange works for me.

**Comment by BenMagyar** [comment]
No worries, released 0.6.0 with `resetForm`.
